### PR TITLE
fix bug for encTypeInt32

### PR DESCRIPTION
**Pull request by fangyincheng**
**What this PR does**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: int_test.go [hunk 1]
```diff
@@ -63,6 +63,28 @@ func TestEncInt32Len2B(t *testing.T) {
 	t.Logf("decode(%#x) = %#x, %v\n", v, res, err)
 }
 
+func TestEncInt32ForAlias(t *testing.T) {
+	type JavaEnum int32
```
Comment (wongoo) [inline int_test.go]: duplicated definition?

File: int_test.go [hunk 2]
```diff
@@ -63,6 +63,28 @@ func TestEncInt32Len2B(t *testing.T) {
...
+
+	v = 0xe6
+	// var v int32 = 0xf016
+	e = NewEncoder()
+	e.Encode(v)
+	if len(e.Buffer()) == 0 {
+		t.Fail()
+	}
+	d = NewDecoder(e.Buffer())
+	res, err = d.Decode()
```
Comment (wongoo) [inline int_test.go]: assert the err is nil & compare the response

File: go.mod [hunk 3]
```diff
@@ -5,3 +5,5 @@ require (
 	github.com/pkg/errors v0.8.1
 	github.com/stretchr/testify v1.4.0
 )
+
+go 1.13
```
Comment (wongoo) [inline go.mod]: remove this

File: int_test.go (hunk 2 above)
Comment (fangyincheng) [reply int_test.go]: done.

File: int.go [hunk 4]
```diff
@@ -109,7 +109,7 @@ func (d *Decoder) decInt32(flag int32) (int32, error) {
 
 func (d *Encoder) encTypeInt32(b []byte, p interface{}) ([]byte, error) {
 	value := reflect.ValueOf(p)
-	if value.IsNil() {
+	if PackPtr(value).IsNil() {
```
Comment (wongoo) [inline int.go]: the param p will always be a pointer, seems not needed if u already check `POJOEnum` first.

Comment (fangyincheng) [reply int.go]: `PackPtr` make other alias be encoded successfully.

**Comment by wongoo** [comment]
LGTM

**Pull request merged**